app/backend/db/schemas.py

```diff
@@ -0,0 +1,86 @@
+from enum import Enum 
+from typing import List, Dict, Any
+from datetime import date, time, datetime
+
+from pydantic import BaseModel
+
+
+class Mador(BaseModel):
+    name: str
+    manager: int = None
+
+    class Config:
+        orm_mode = True
+
+
+class User(BaseModel):
+    id: int
+    commander: int = None
+    reminder_time: time = None
+    english_name: str = None
+    mador: Mador = None
+    operators_id: List[int] = None
+
+    class Config:
+        orm_mode = True
+
+
+class Hierarchy(BaseModel):
+    """"Hierarchy for User and Commander."""
+    hierarchy: Dict[User, Any] = None
+
+
+class HierarchyList(BaseModel):
+    """Hierarchy for Reporter and Admin 
+        that can view more then 1 hierarchy.
+    """
+    hierarchys: List[Hierarchy]
+
+
+class StatusTypes(str, Enum):
+    required = "required"
+    not_required = "not_required"
+    not_important = "not_important"
+
+
+class DateDetails(BaseModel):
+    type: StatusTypes = None
+
+
+class AnswerStateTypes(str, Enum):
+    here = "here"
+    not_here = "not_here"
+
+
+class DateDataBody(BaseModel):
+    user_id: int 
+    start_date: date
+    end_date: date
+    state: AnswerStateTypes
+    reason: str = None
+    reported_by_id: int
+    reported_time: datetime
+
+    class Config:
+        orm_mode = True
+
+
+class DateDataResponse(BaseModel):
+    user_id: int 
+    date: date
```

Comment:
in the response we should give date details object

Reply:
Got it